src/hotspot/share/runtime/continuationFreezeThaw.cpp

```diff
@@ -1619,15 +1619,15 @@ static int num_java_frames(ContinuationWrapper& cont) {
 }
 
 static void invalidate_jvmti_stack(JavaThread* thread) {
-  if (thread->is_interp_only_mode()) {
+  if (JvmtiExport::can_post_frame_pop() || thread->is_interp_only_mode()) {
```

Comment:
Can you please explain why this change is required? Doesn't 'invalidate_cur_stack_depth' make sense only when interp_only mode is enabled for the threads only?
It is invalidated once thread switched to interp only.

Reply:
> The interp_only check is needed for optimization to avoid a performance overhead of current stack depth invalidation.
But if we are not in interp_only mode isn't it already invalidated?